### PR TITLE
Hive config properties docs improvements

### PR DESCRIPTION
**Pull request by colebow**
## Description

Refactor and minor grammar edits of various tables in the Hive docs page, along with adding some missing config properties to a couple tables.

## Additional context and related issues

Some of the work in the final commit was lifted from #10831, authorship credit is in commit message.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by m57lyra** [comment]
@m57lyra self-tag

File: docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -607,49 +615,92 @@ Specific properties can be used to further configure the
...
+     - SOCKS proxy to use for the Thrift Hive metastore.
+     -
+   * - ``hive.metastore.thrift.client.max-retries``
+     - Maximum number of retry attempts for metastore requests.
+     - ``9``
+   * - ``hive.metastore.thrift.client.backoff-scale-factor``
+     - Scale factor for metastore request retry delay.
+     - ``2.0``
+   * - ``hive.metastore.thrift.client.max-retry-time``
+     - Total time limit for a metastore request to be retried.
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/hive.rst:693]: optional: `Total time limit allowed for a metastore request to be retried.`

**Review by findinpath — APPROVED** [review]

**Review by mosabua — COMMENTED** [review]
In general we typically do not want a separate columns for the default value since it looks bad in many cases when only a few values per table actually have a default value. 

Also if we change the column titles to be sentence case (as they should be), we probably want to do that as a separate PR for all docs.

Beyond that .. this is great.

**Comment by colebow** [comment]
Closing this PR, as @m57lyra is going to ingesting the big changes into #15637 so we don't have a giant merge conflict fiesta.

**Pull request closed without merging**